### PR TITLE
Upgrade Rust 1.75.0 nightly

### PR DESCRIPTION
**Pull request by jgreat**
Upgrade rust version and resolve new lint/clippy requirements 

Highlights:
`clippy::get-first` 
use `first()` instead of `get(0)`

`clippy::unnecessary_fallible_conversions` 
Use `from()` and `into()` instead of `try_from()` and `try_into()`.  This one might need more work. We loose the `.expect()` or `map_err()` so some custom error messages.  

`clippy::manual_hash_one`
places where we could use `hash_one` method instead of longer `hash`, `hasher.finish()` style.

Clean up some unused imports.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -9078,9 +9078,9 @@ checksum = "dcf81ac59edc17cc8697ff311e8f5ef2d99fcbd9817b34cec66f90b6c3dfd987"
 
 [[package]]
 name = "ubyte"
-version = "0.10.1"
+version = "0.10.4"
```
Comment (jgreat) [inline Cargo.lock:9081]: update `ubyte` for cargo build version deprecation notice.

File: api/src/convert/amount.rs [hunk 2]
```diff
@@ -149,7 +149,7 @@ mod tests {
         let amount = Amount::new(10000, TokenId::from(10));
 
         let external_amount: external::Amount = (&amount).into();
-        let recovered_amount: Amount = (&external_amount).try_into().unwrap();
+        let recovered_amount: Amount = (&external_amount).into();
```
Comment (jgreat) [inline api/src/convert/amount.rs:152]: clippy::unnecessary_fallible_conversions

File: api/src/convert/block_metadata.rs [hunk 3]
```diff
@@ -34,11 +34,11 @@ impl TryFrom<&blockchain::BlockMetadataContents> for BlockMetadataContents {
         let quorum_set = src.get_quorum_set().try_into()?;
         let attestation_evidence = match &src.attestation_evidence {
             Some(BlockMetadataContents_oneof_attestation_evidence::dcap_evidence(evidence)) => {
-                let evidence = evidence.try_into()?;
+                let evidence = evidence.into();
```
Comment (jgreat) [inline api/src/convert/block_metadata.rs:37]: clippy::unnecessary_fallible_conversions

File: consensus/scp/src/ballot.rs [hunk 4]
```diff
@@ -62,9 +62,8 @@ impl<V: Value> PartialOrd for Ballot<V> {
 // This makes debugging easier when looking at large ballots.
 impl<V: Value> fmt::Display for Ballot<V> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let mut hasher = HasherBuilder::default().build_hasher();
-        self.X.hash(&mut hasher);
-        let hashed_X_values = hasher.finish();
+        let hasher = HasherBuilder::default();
+        let hashed_X_values = hasher.hash_one(&self.X);
```
Comment (jgreat) [inline consensus/scp/src/ballot.rs:66]: Resolving `clippy::manual_hash_one`

File: consensus/scp/src/msg.rs [hunk 5]
```diff
@@ -613,11 +613,7 @@ impl<V: Value, ID: GenericNodeId> fmt::Display for Msg<V, ID> {
         // Returns "<set.len, hash(set)>".
         let hasher_builder = HasherBuilder::default();
         let format_b_tree_set = |b_tree_set: &BTreeSet<V>| {
-            let hash = {
-                let mut hasher = hasher_builder.build_hasher();
-                b_tree_set.hash(&mut hasher);
-                hasher.finish()
-            };
+            let hash = hasher_builder.hash_one(b_tree_set);
```
Comment (jgreat) [inline consensus/scp/src/msg.rs:616]: Resolving `clippy::manual_hash_one`

File: consensus/scp/types/src/quorum_set.rs [hunk 6]
```diff
@@ -234,24 +234,16 @@ impl<ID: GenericNodeId + AsRef<ResponderId>> From<&QuorumSet<ID>> for QuorumSet<
...
-            let mut hasher = hasher_builder.build_hasher();
-            quorum_set_1.hash(&mut hasher);
-            hasher.finish()
-        };
-        let quorum_set_2_hash = {
-            let mut hasher = hasher_builder.build_hasher();
-            quorum_set_2.hash(&mut hasher);
-            hasher.finish()
-        };
+        let quorum_set_1_hash = hasher_builder.hash_one(quorum_set_1);
```
Comment (jgreat) [inline consensus/scp/types/src/quorum_set.rs:245]: Resolving `clippy::manual_hash_one`

File: sgx/compat/src/lib.rs [hunk 7]
```diff
@@ -13,9 +13,7 @@ cfg_if! {
         pub use mc_sgx_panic as panic;
 
         // Compat with std
-        mod thread {
-            pub use mc_sgx_panic::thread_panicking as panicking;
-        }
+        mod thread {}
```
Comment (jgreat) [inline sgx/compat/src/lib.rs:16]: These `use` were coming up as an unused imports.

File: transaction/core/tests/input_rules.rs [hunk 8]
```diff
@@ -13,10 +13,9 @@ use mc_transaction_core::{
 use mc_util_from_random::FromRandom;
 use mc_util_test_helper::{RngType, SeedableRng};
 
-use util::{
-    create_test_tx, create_test_tx_with_amount_and_comparer_and_recipients,
-    INITIALIZE_LEDGER_AMOUNT,
-};
+use util::{create_test_tx, create_test_tx_with_amount_and_comparer_and_recipients};
+
+use mc_ledger_db::test_utils::INITIALIZE_LEDGER_AMOUNT;
```
Comment (jgreat) [inline transaction/core/tests/input_rules.rs:18]: I moved this this `use` from `transaction/core/tests/util/mod.rs`. This seems OK to me but would love a doublecheck this that this is what we intend.

File: util/grpc/src/lib.rs [hunk 9]
```diff
@@ -39,7 +39,6 @@ pub use crate::{
         BasicCredentials, TokenAuthenticator, TokenBasicCredentialsGenerator,
         TokenBasicCredentialsGeneratorError, ANONYMOUS_USER,
     },
-    autogenerated_code::*,
```
Comment (jgreat) [inline util/grpc/src/lib.rs:42]: removed because it was listed as unused.  Did this do anything special?

Comment (nick-mobilecoin) [reply util/grpc/src/lib.rs:42]: it's a duplicate from line 21, so is uneeded

**Review by nick-mobilecoin — APPROVED** [review]
This is going to change the enclaves
I think this change should probably target `main`
Other than that looks good

**Comment by nick-mobilecoin** [comment]
Also the PR title says "Upgrade Rust 1.75.0 nightly" but the cargo.toml files still have
```
[workspace.package]
rust-version = "1.74.0"
```
Should probably update that to 1.75.0 as well, though that might need other changes so maybe it should get its own PR

**Pull request merged**